### PR TITLE
Test/lib mediator merge

### PR DESCRIPTION
**Pull request by jendis**
## Description
The first bunch of test scenarios for libMediator with extension of libTestUtils.

## Status
Several test scenarios do just a execution of methods for increase of coverage without testing of correct behavior because of void return value. In this case application of mocking for outer classes would be more appropriate and it would significantly increase the quality of tests.

### Implementation

- [x] **ready for review**

### Integration Test (Core Team)

- [x] local machine test

- [ ] small-scale cloud test

### CODE REVIEW

File: tests/libTestUtils/TestUtils.h [hunk 1]
```diff
@@ -45,15 +46,26 @@ uint8_t Dist1to99();
...
 MicroBlockHeader GenerateRandomMicroBlockHeader();
 TxBlockHeader GenerateRandomTxBlockHeader();
 VCBlockHeader GenerateRandomVCBlockHeader();
 FallbackBlockHeader GenerateRandomFallbackBlockHeader();
 CoSignatures GenerateRandomCoSignatures();
+
+using DS_Comitte_t = std::deque<std::pair<PubKey, Peer>>;
+DS_Comitte_t GenerateRandomDSCommittee(uint32_t);
+
+Shard generateRandomShard(size_t);
```
Comment (ansnunez) [inline tests/libTestUtils/TestUtils.h]: Please maintain Pascal case for the function names

File: tests/libTestUtils/TestUtils.cpp [hunk 2]
```diff
@@ -150,5 +174,27 @@ FallbackBlockHeader GenerateRandomFallbackBlockHeader() {
...
+  DS_Comitte_t ds_c;
+  for (uint32_t i = 1; i <= size; i++) {
+    ds_c.push_front(std::make_pair(GenerateRandomPubKey(), Peer()));
+  }
+  return ds_c;
+}
+
+Shard generateRandomShard(size_t size) {
+  Shard s;
+  for (size_t i = 1; i <= size; i++)
```
Comment (ansnunez) [inline tests/libTestUtils/TestUtils.cpp]: For consistency, please put for loop statements within curly braces, even if only one statement is contained.

File: tests/libTestUtils/TestUtils.h [hunk 3]
```diff
@@ -45,15 +46,26 @@ uint8_t Dist1to99();
 uint8_t DistUint8();
 uint16_t DistUint16();
 uint32_t DistUint32();
+uint32_t DistUint64();
+uint32_t DistUint128();
```
Comment (ansnunez) [inline tests/libTestUtils/TestUtils.h]: This seems unimplemented.

File: tests/libTestUtils/TestUtils.h (hunk 1 above)
Comment (jendis) [reply tests/libTestUtils/TestUtils.h]: Fixed - sorry again.

File: tests/libTestUtils/TestUtils.cpp (hunk 2 above)
Comment (jendis) [reply tests/libTestUtils/TestUtils.cpp]: Done

File: tests/libTestUtils/TestUtils.h (hunk 3 above)
Comment (jendis) [reply tests/libTestUtils/TestUtils.h]: Copy/paste issue. 
uint32_t DistUint128() - removed, I wasn't able to find numerical limit for 128bit integer

File: tests/libTestUtils/TestUtils.h [hunk 4]
```diff
@@ -45,15 +46,26 @@ uint8_t Dist1to99();
 uint8_t DistUint8();
 uint16_t DistUint16();
 uint32_t DistUint32();
+uint32_t DistUint64();
```
[the remark this answers is not in the document]
Comment (jendis) [reply tests/libTestUtils/TestUtils.h]: Copy/paste issue.
Fixed.

File: tests/Mediator/constants.xml [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+        <OUTPUT_JSON>output.json</OUTPUT_JSON>
+        <INPUT_CODE>input.scilla</INPUT_CODE>
+    </smart_contract>
+    <accounts>
+        <account>
+            <wallet_address>cc02a3c906612cc5bdb087a30e6093c9f0aa04fc</wallet_address>
+        </account>
+    </accounts>
+    <lookups>
+        <!--IP to be provided after public testnet launch.
```
Comment (Jazzz42) [inline tests/Mediator/constants.xml]: I would like to discuss this part about providing IP. The process has to be tested too.

**Comment by jendis** [comment]
Commit blow-up. Created new branch test/libMediator to avoid possible time consuming rebase from which new pull request will be created.
This branch test/libMediator_merge is going to be deleted.

**Pull request closed without merging**